include/mbgl/map/map.hpp

```diff
@@ -182,10 +183,11 @@ class Map : private util::noncopyable {
     double getDefaultPitch() const;
 
     // Feature queries
-    std::vector<Feature> queryRenderedFeatures(const ScreenCoordinate&, const QueryOptions& options = {});
-    std::vector<Feature> queryRenderedFeatures(const ScreenBox&,        const QueryOptions& options = {});
-    
+    std::vector<Feature> queryRenderedFeatures(const ScreenCoordinate&, const RenderedQueryOptions& options = {});
+    std::vector<Feature> queryRenderedFeatures(const ScreenBox&,        const RenderedQueryOptions& options = {});
+
     AnnotationIDs queryPointAnnotations(const ScreenBox&);
+    std::vector<Feature> querySourceFeatures(const std::string&,        const SourceQueryOptions& options = {});
```

Comment:
> This does mean that you have to retrieve the source in the SDKs before querying. Small overhead, but still. Is this what we want?

Yes, I think so -- by eliminating the `map` middleman in `map.querySourceFeatures(source)`, `source.queryFeatures()` becomes conceptually simpler and possibly actually _more_ efficient in the case of repeated queries (though I would indeed expect the benefit to be negligible).

> shall I move SourceQueryOptions to a separate header in the `mbgl::style` namespace?

Yeah, I think that makes sense.